### PR TITLE
Internet traffic to Palo Alto with some additional fixes

### PR DESCRIPTION
**Pull request by thomast1906**
### Change description ###
- Internet traffic to Palo Alto
- Demo environment not included for now
- Swapped route table to array/copy for ease of use, in future it is planned to add additional routes for private IP routing
- Removed ITHC hub reference as it has another route assigned with different name
- Added routetableName parameter due to ITHC having a different route table assigned and private IP range

Raising PR for review, all envs applied to except for Production Jenkins & Production. Will merge once this is complete

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: templates/arm-vnet-template.json [hunk 1]
```diff
@@ -118,8 +118,21 @@
     "serviceEndpoints": {
       "type": "array",
       "defaultValue": []
+    },
+    "routetableName": {
+      "type": "string",
+      "metadata": {
+        "description": "The full ID of the hub vnet to peer to"
```
Comment (timja) [inline templates/arm-vnet-template.json:125]: is this description right?

File: templates/vars/vnet/ithc.json [hunk 2]
```diff
@@ -51,6 +51,31 @@
           "service": "Microsoft.Storage"
         }
       ]
-    }
+    },
+    "routetableName": {
+      "value": "aks-ithc-rt"
```
Comment (timja) [inline templates/vars/vnet/ithc.json:56]: better to fix this than hack it

File: templates/arm-vnet-template.json [hunk 3]
```diff
@@ -118,8 +118,21 @@
     "serviceEndpoints": {
       "type": "array",
       "defaultValue": []
+    },
+    "routetableName": {
```
Comment (timja) [inline templates/arm-vnet-template.json:122]: shouldn't be a parameter imo better to fix ithc

File: templates/arm-vnet-template.json (hunk 1 above)
Comment (thomast1906) [reply templates/arm-vnet-template.json:125]: Will remove parameter as below

File: templates/vars/vnet/ithc.json (hunk 2 above)
Comment (thomast1906) [reply templates/vars/vnet/ithc.json]: Cheers - now updated & deployed with aks-route-table

File: templates/arm-vnet-template.json (hunk 3 above)
Comment (thomast1906) [reply templates/arm-vnet-template.json]: Cheers - now updated

File: templates/vars/vnet/aat.json [hunk 4]
```diff
@@ -66,6 +66,22 @@
           "service": "Microsoft.Storage"
         }
       ]
-    }
+    },
+    "routetableRoutes": {
+      "value": [
+          {
+          "name": "BastionMgmtVnet",
```
Comment (timja) [inline templates/vars/vnet/aat.json:73]: is this actually needed? since you've got the hub route in and you're sending all traffic there?

File: templates/vars/vnet/prod.json [hunk 5]
```diff
@@ -51,6 +51,28 @@
           "service": "Microsoft.Storage"
         }
       ]
-    }
+    },
+    "bastionSubnetPrefix": {
+      "value": "10.48.0.0/27"
+    },
+    "paloaltoIPNextHop": {
```
Comment (timja) [inline templates/vars/vnet/prod.json:58]: why is this a parameter?

File: templates/vars/vnet/prod.json [hunk 6]
```diff
@@ -51,6 +51,28 @@
           "service": "Microsoft.Storage"
         }
       ]
-    }
+    },
+    "bastionSubnetPrefix": {
```
Comment (timja) [inline templates/vars/vnet/prod.json:55]: and this?

File: templates/vars/vnet/prod.json [hunk 7]
```diff
@@ -51,6 +51,28 @@
...
+      "value": "10.48.0.0/27"
+    },
+    "paloaltoIPNextHop": {
+      "value": "10.11.8.36"
+    },
+    "routetableRoutes": {
```
Comment (timja) [inline templates/vars/vnet/prod.json:61]: I thought prod was being left out of this PR?

Comment (timja) [reply templates/vars/vnet/prod.json:61]: ah i see you're merging once prod is done

Comment (thomast1906) [reply templates/vars/vnet/prod.json]: Going to leave Prod out of this for now, its not due to be applied until 12th April

File: templates/arm-vnet-template.json [hunk 8]
```diff
@@ -118,11 +118,19 @@
     "serviceEndpoints": {
       "type": "array",
       "defaultValue": []
+    },
+    "routetableRoutes": {
+      "type": "array",
+      "metadata": {
+        "description": "The address prefix of the bastion subnet"
```
Comment (timja) [inline templates/arm-vnet-template.json:125]: looks wrong?

File: templates/vars/vnet/aat.json (hunk 4 above)
Comment (timja) [reply templates/vars/vnet/aat.json:78]: this seems an issue?

File: templates/vars/vnet/cftptl-intsvc.json [hunk 9]
```diff
@@ -129,6 +129,22 @@
...
+    "routetableRoutes": {
+      "value": [
+          {
+          "name": "Internet",
+          "nextHopType": "VirtualAppliance",
+          "nextHopIpAddress": "10.11.8.36",
+          "addressPrefix": "0.0.0.0/0"
+        },
+        {
+          "name": "sdsdev",
```
Comment (timja) [inline templates/vars/vnet/cftptl-intsvc.json:142]: why is this needed? won't the hub route catch this?

File: templates/vars/vnet/cftsbox-intsvc.json [hunk 10]
```diff
@@ -51,6 +51,23 @@
           "service": "Microsoft.Storage"
         }
       ]
-    }
+    },
+    "routetableRoutes": {
+      "value": [
+          {
+          "name": "BastionMgmtVnet",
```
Comment (timja) [inline templates/vars/vnet/cftsbox-intsvc.json:58]: is this needed?

File: templates/vars/vnet/ithc.json [hunk 11]
```diff
@@ -3,7 +3,7 @@
   "contentVersion": "1.0.0.0",
   "parameters": {
     "hubVnetId": {
-      "value": "/subscriptions/fb084706-583f-4c9a-bdab-949aac66ba5c/resourceGroups/hmcts-hub-nonprodi/providers/Microsoft.Network/virtualNetworks/hmcts-hub-nonprodi"
```
Comment (timja) [inline templates/vars/vnet/ithc.json:6]: why this change?

File: templates/vars/vnet/ithc.json [hunk 12]
```diff
@@ -56,6 +56,28 @@
           "service": "Microsoft.Storage"
         }
       ]
-    }
+    },
+    "routetableRoutes": {
+      "value": [
+          {
+          "name": "AKSITHCtoHUBPalo",
```
Comment (timja) [inline templates/vars/vnet/ithc.json:63]: why is this route here twice?

File: templates/vars/vnet/ithc.json [hunk 13]
```diff
@@ -56,6 +56,28 @@
...
+          "nextHopType": "VirtualAppliance",
+          "nextHopIpAddress": "10.11.72.36",
+          "addressPrefix": "10.0.0.0/8"
+          },
+          {
+            "name": "BastionMgmtVnet",
```
Comment (timja) [inline templates/vars/vnet/ithc.json:69]: ?

File: templates/vars/vnet/perftest.json [hunk 14]
```diff
@@ -56,6 +56,22 @@
           "service": "Microsoft.Storage"
         }
       ]
-    }
+    },
+    "routetableRoutes": {
+      "value": [
+          {
+            "name": "BastionMgmtVnet",
```
Comment (timja) [inline templates/vars/vnet/perftest.json:63]: ?

File: templates/vars/vnet/preview.json [hunk 15]
```diff
@@ -76,7 +76,22 @@
           "service": "Microsoft.Storage"
         }
       ]
-    }
+    },
+    "routetableRoutes": {
+      "value": [
+          {
+            "name": "BastionMgmtVnet",
```
Comment (timja) [inline templates/vars/vnet/preview.json:83]: ?

File: templates/vars/vnet/sbox.json [hunk 16]
```diff
@@ -61,6 +61,23 @@
           "service": "Microsoft.Storage"
         }
       ]
-    }
+    },
+    "routetableRoutes": {
+      "value": [
+          {
+          "name": "BastionMgmtVnet",
```
Comment (timja) [inline templates/vars/vnet/sbox.json:68]: ?

**Review by timja — CHANGES_REQUESTED** [review]
few questions

File: templates/vars/vnet/ithc.json (hunk 11 above)
Comment (thomast1906) [reply templates/vars/vnet/ithc.json]: reverted, some reason thought ops-peerings covered this now.

File: templates/arm-vnet-template.json (hunk 8 above)
Comment (thomast1906) [reply templates/arm-vnet-template.json]: Updated dsc. copy fail

File: templates/vars/vnet/cftptl-intsvc.json (hunk 9 above)
Comment (thomast1906) [reply templates/vars/vnet/cftptl-intsvc.json:142]: Was requested by Luigi

File: templates/vars/vnet/ithc.json (hunk 11 above)
Comment (timja) [reply templates/vars/vnet/ithc.json]: i think it only does ukwest, need to get rid of it from ops-peering but can do that when you move to terraform

File: templates/vars/vnet/cftptl-intsvc.json (hunk 9 above)
Comment (timja) [reply templates/vars/vnet/cftptl-intsvc.json:142]: @luigibk why is the specific route needed?

File: templates/vars/vnet/ithc.json (hunk 13 above)
Comment (thomast1906) [reply templates/vars/vnet/ithc.json:63]: Will remove these BastionMgmtVnet for now, they were added I assume manually at some stage prior.

File: templates/vars/vnet/cftptl-intsvc.json (hunk 9 above)
Comment (luigibk) [reply templates/vars/vnet/cftptl-intsvc.json:142]: Grafana

Comment (timja) [reply templates/vars/vnet/cftptl-intsvc.json:142]: won't it work anyway once it uses a default route to the hub?

Comment (luigibk) [reply templates/vars/vnet/cftptl-intsvc.json:142]: Until it's overridden by something propagated from somewhere as we've seen quite often recently ... yes

Comment (timja) [reply templates/vars/vnet/cftptl-intsvc.json:142]: can we remove and only add if required then? it would be better to rely on default route to hub as much as possible imo

Comment (luigibk) [reply templates/vars/vnet/cftptl-intsvc.json:142]: That depends really, but remove it for now and it can always be added again later

Comment (thomast1906) [reply templates/vars/vnet/cftptl-intsvc.json:142]: Cheers both - now removed.

File: templates/vars/vnet/aat.json (hunk 4 above)
Comment (thomast1906) [reply templates/vars/vnet/aat.json]: Removed BastionMgmtVnet reference(s)

**Review by timja — APPROVED** [review]
